src/server/pkg/errutil/errutil.go

```diff
@@ -30,3 +30,20 @@ func IsNotFoundError(err error) bool {
 	}
 	return strings.Contains(err.Error(), "not found")
 }
+
+// IsInvalidNameError returns true if err is due to an invalid name
+func IsInvalidNameError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), "only alphanumeric characters, underscores, and dashes are allowed")
+}
+
+// IsHasNoHeadError returns true if the err is due to an operation that cannot
+// be performed on a headless branch
+func IsHasNoHeadError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), "has no head")
+}
```

Comment:
This error is defined in `src/server/pfs/pfs.go` along with a bunch of error predicates for detecting the other errors in that file. I'd add this predicate there, rather than here so it'll get updated if the error changes.